flink-runtime/src/test/java/org/apache/flink/runtime/scheduler/DefaultSchedulerTest.java

```diff
@@ -175,7 +173,7 @@ public class DefaultSchedulerTest extends TestLogger {
     @BeforeEach
     void setUp() {
         executor = Executors.newSingleThreadExecutor();
-        scheduledExecutorService = new DirectScheduledExecutorService();
+        scheduledExecutorService = Executors.newSingleThreadScheduledExecutor();
```

Comment:
The test would get stuck otherwise due to the assertRunningInMainThread check (this was a bug that was hidden by waiting for time out on `taskManagerGateway.waitForSubmissions(1);`)